packages/nimble-components/src/wafer-map/modules/zoom-handler.ts

```diff
@@ -0,0 +1,122 @@
+import { select } from 'd3-selection';
+import {
+    zoom,
+    ZoomBehavior,
+    zoomIdentity,
+    ZoomTransform,
+    zoomTransform
+} from 'd3-zoom';
+import type { DataManager } from './data-manager';
+import type { RenderingModule } from './rendering';
+
+/**
+ * ZoomHandler deals with user interactions and events like zooming
+ */
+export class ZoomHandler {
+    private zoomTransform: ZoomTransform = zoomIdentity;
+    private readonly minScale = 1.1;
+    private readonly minExtentPoint: [number, number] = [-100, -100];
+    private readonly extentPadding = 100;
+
+    public constructor(
+        private readonly canvas: HTMLCanvasElement,
+        private readonly zoomContainer: HTMLElement,
+        private readonly dataManager: DataManager,
+        private readonly renderingModule: RenderingModule
+    ) {}
+
+    public attachZoomBehaviour(): void {
```

Comment:
I'm Canadian so I can handle the British spelling of behavior/behaviour but the rest of the American dev team will probably get confused 😂. (Adding a u is also inconsistent with the naming elsewhere in the file)
```suggestion
    public attachZoomBehavior(): void {
```

Reply:
I will fix 😂😂